### PR TITLE
fix: alphanumeric check on older browsers doesn't throw

### LINKED ISSUE #2430: Fails with a regular expression error in older browsers
**Marked version:** 4.0.12

**Describe the bug**
- Fails with an error `invalid identity escape in regular expression` on older browsers (Firefox <= 72)

Снимок экрана 2022-03-30 в 14 15 14

**To Reproduce**

**Expected behavior**

### PR DESCRIPTION
**Pull request by sheremet-va**
Fixes #2430

**Marked version:** 4.0.12

**Markdown flavor:** n/a

## Description

- Fixes an error on older browsers when doing alphanumeric check. Right now it throws an error `invalid identity escape in regular expression`

Снимок экрана 2022-03-30 в 14 15 14

I think `\w` should be enough for a fallback. I wanted to to `\uXXX-\uXXX` for a regexp, but it was too long: https://util.unicode.org/UnicodeJsps/regex.jsp?a=%5Cp%7BL%7D%5Cp%7BN%7D&b=

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -247,3 +247,14 @@ export function repeatString(pattern, count) {
   }
   return result + pattern;
 }
+
+export function isAlphanumeric(text) {
+  // \p cases older browsers (Firefox <= 72) to throw an error
```
Comment (UziTech) [inline src/helpers.js:252]:
```suggestion
  // \p causes older browsers (Firefox <= 72) to throw an error
```

**Review by UziTech — CHANGES_REQUESTED** [review]

**Comment by UziTech** [comment]
Looks like this affects about 7% of people. https://caniuse.com/mdn-javascript_builtins_regexp_property_escapes. The only thing I'm concerned about is without a test this could easily be broken again in the future without us knowing about it.

One alternative could be to use a [babel plugin](https://babeljs.io/docs/en/babel-plugin-proposal-unicode-property-regex) to convert this in the `umd` and/or `min` versions.

**Comment by calculuschild** [comment]
I was gonna say... I think we should leverage Babel for compatibility issues. Building our own polyfills/workarounds for old browsers seems outside the scope of Marked, plus a pain to test and maintain.

**Comment by UziTech** [comment]
I'm going to close this as it is already fixed.

**Pull request closed without merging**